### DISCUSSION TITLE
redis full of keys with format my-custom-queue-name:id

### DISCUSSION THREAD

**Discussion by yongzhang** [OptimalBits/bull#2632] [General]
## Description

Hi, 

As described in the title, I can see there're a large number of keys in redis with format `my-custom-queue-name:id`, although I added args:
```
removeOnComplete: true
removeOnFailed: true
```

Can anyone explain what are those keys and how to clean up? Thanks.

## Bull version

v4.10.4

**Comment by manast** [comment]
Do you have a concrete example that produces that behavior?

**Comment by yongzhang** [comment]
> Do you have a concrete example that produces that behavior?

Not sure if this is a good example here, I monitored keys in redis:

Job Start:
```
2) "order:20230725091940ww81ghyJc0oJ-queue-b38f7e2e-7249-4736-9208-a051adb789b3:id"
3) "order:20230725091940ww81ghyJc0oJ-queue-b38f7e2e-7249-4736-9208-a051adb789b3:2"
4) "order:20230725091940ww81ghyJc0oJ-queue-b38f7e2e-7249-4736-9208-a051adb789b3:2:lock"
5) "order:20230725091940ww81ghyJc0oJ-queue-b38f7e2e-7249-4736-9208-a051adb789b3:stalled-check"
6) "order:20230725091940ww81ghyJc0oJ-queue-b38f7e2e-7249-4736-9208-a051adb789b3:active"
```

Final:
```
2) "order:20230725091940ww81ghyJc0oJ-queue-b38f7e2e-7249-4736-9208-a051adb789b3:id"
3) "order:20230725092004sDyxcjGqpcsp-queue-f5c70d5b-8a41-4dff-89ab-850a624bb439:id"
```

Keys with other suffix were removed from redis automatically, but not with ":id" suffix. Now there're huge of them left in redis.

**Comment by manast** [comment]
The id key keeps the job counter and is necessary for a queue. The question is if you are using an unbounded number of queues, which is not the way you should use the library.
